vscode-dotnet-runtime-library/src/Acquisition/ExistingPathResolver.ts

```diff
@@ -36,21 +66,121 @@ export class ExistingPathResolver
                 {
                     const existingLocalPath = existingPaths.individualizedExtensionPaths?.filter((pair) => pair.extensionId === extensionId);
                     if (existingLocalPath && existingLocalPath.length > 0) {
-                        return { dotnetPath: existingLocalPath![0].path };
+                        return existingLocalPath![0].path;
                     }
                 }
                 else if (existingPaths.sharedExistingPath)
                 {
-                    return { dotnetPath: existingPaths.sharedExistingPath}
+                    return existingPaths.sharedExistingPath;
                 }
                 else
                 {
                     windowDisplayWorker.showWarningMessage(
                         `Ignoring existing .NET paths defined in settings.json because the setting is only set for other extensions, and not for ${extensionId}`,
                         () => { /* No callback */ },
                     );
+                    return null;
                 }
             }
         }
+
+        return null;
+    }
+
+    private allowInvalidPath(workerContext : IAcquisitionWorkerContext) : boolean
+    {
+        return workerContext.allowInvalidPathSetting ?? false;
+    }
+
+    private async providedPathMeetsAPIRequirement(workerContext : IAcquisitionWorkerContext, existingPath : string, apiRequest : IDotnetAcquireContext) : Promise<boolean>
+    {
+
+        const availableRuntimes = await this.getRuntimes(existingPath);
+        const requestedMajorMinor = versionUtils.getMajorMinor(apiRequest.version, this.workerContext.eventStream, this.workerContext);
+
+        if(availableRuntimes.some((runtime) =>
+        {
+            return runtime.mode === apiRequest.mode && versionUtils.getMajorMinor(runtime.version, this.workerContext.eventStream, this.workerContext) === requestedMajorMinor;
+        }))
+        {
+            return true;
+        }
```

Comment:
nit: This formatting seems a little odd to me. I think I'd expect the first set of `{}` to be intended and not directly under the `if` width. That way, it doesn't look like it is the body of the if-statement. Same thing for `if(availableSDKs.some` a few lines below this.

Reply:
That's a good point, I will fix this and the other indent issue here: https://github.com/dotnet/vscode-dotnet-runtime/pull/1954

I've refactored the code out into it's own file without any changes so it will be easier to review. Since this file will be deleted, it doesn't make sense to fix it in this PR.